diff --git a/cmd/portunus-orchestrator/ldap.go b/cmd/portunus-orchestrator/ldap.go
index 9564c5e..40cd2d7 100644
--- a/cmd/portunus-orchestrator/ldap.go
+++ b/cmd/portunus-orchestrator/ldap.go
@@ -134,7 +134,7 @@ func runLDAPServer(environment map[string]string) {
 
 	bindURL := "ldap:///"
 	if environment["PORTUNUS_SLAPD_TLS_CERTIFICATE"] != "" {
-		bindURL = "ldaps:///"
+		bindURL = "ldap:/// ldaps:///"
 	}
 
 	logg.Info("starting LDAP server")
